aether-kernel/aether/kernel/api/entity_extractor.py

```diff
@@ -120,10 +120,9 @@ def extract_view(request, *args, **kwargs):
 
     delta = request.query_params.get('delta', '1d')
     modified = parse_delta(delta)
-    submissions = Submission.objects.filter(is_extracted=False, modified__lte=modified)
-    count = submissions.count()
-
-    for submission in submissions:
+    submissions = Submission.objects.filter(modified__lte=modified).filter(is_extracted=False)
+    count = 0
+    for count, submission in enumerate(submissions):
         send_model_item_to_redis(submission)
```

Comment:
```suggestion
    submissions = Submission.objects.filter(modified__lte=modified).filter(is_extracted=False)
    count = submissions.count()

    #  choose a meaningful page size
    page_size = 20
    pages = range(0, 1 + count / page_size)  # not completely sure
    for page in pages:
        _start = page_size * page
        _end = page_size * (page + 1) - 1
        if _end > count: _end = count

        for submission in submissions[_start:_end]:
            send_model_item_to_redis(submission)
```

enumerate breaks the performance, it's getting all the records in one call. 

I didn't check the code but it's more or less what I did for the export task